core/signingrate/dynamo_signing_rate_storage.go

```diff
@@ -0,0 +1,461 @@
+//     * Allows us to store/retrieve specific buckets efficiently
+//   - Other Attributes:
+//     * EndTimestamp: When the bucket ended
+//     * Payload: The serialized protobuf data (the actual bucket contents)
+//     * PayloadType: A dummy constant value (used as a dummy partition key, explained below)
+//
+// **Global Secondary Index (EndTimestampIndex):**
+//   - Partition Key: PayloadType (always set to "Payload" - a constant dummy value)
+//   - Sort Key: EndTimestamp (allows range queries like "EndTimestamp > X")
+//
+// ## Why This Design?
+//
+// **Problem**: We need to query "all buckets where EndTimestamp > X" to load historical data.
+//   - We can't use the main table because its key is StartTimestamp
+//   - DynamoDB won't let us query by EndTimestamp without an index
+//
+// **Solution**: Create a Global Secondary Index with EndTimestamp as the sort key.
+//   - But GSIs require a partition key (DynamoDB rule)
+//   - We don't have a natural partition key for this query pattern
+//
+// **The Dummy Partition Key Trick**:
+//   - We create an artificial attribute called PayloadType that's always "Payload"
+//   - Every item gets the same PayloadType value
+//   - This puts all items in the same partition for the GSI
+//   - Now we can query: "PayloadType = 'Payload' AND EndTimestamp > X"
+//
+// **Why Zero-Pad Timestamps?**
+//   - DynamoDB sorts strings lexicographically (like dictionary order)
+//   - "9" > "10" in string comparison, but 9 < 10 numerically
+//   - Zero-padding ensures string sort order matches numerical order
+//   - "0009" < "0010" (correct!)
+//   - We pad to 20 digits to handle the full uint64 range
+//
+// ## Example Query Flow
+//
+// To load all buckets ending after time T:
+//   1. Query the EndTimestampIndex GSI
+//   2. Condition: PayloadType = "Payload" AND EndTimestamp > timestampToString(T)
+//   3. DynamoDB returns matching items ordered by EndTimestamp
+//   4. We deserialize the Payload attribute to get the bucket objects
+//   5. Sort by StartTimestamp for deterministic ordering (EndTimestamp may not be unique)
+//
+// ## Data Format
+//
+// Each item in the table looks like:
+//   {
+//     "StartTimestamp": "00000000001234567890",  // Primary key (zero-padded)
+//     "EndTimestamp":   "00000000001234568890",  // Used for range queries (zero-padded)
+//     "PayloadType":    "Payload",               // Dummy partition key (always "Payload")
+//     "Payload":        <binary protobuf data>   // Serialized SigningRateBucket
+//   }
+//
+// ═══════════════════════════════════════════════════════════════════════════════════════════
+
+const (
+	// DynamoDB attribute names - these define the column names in our table
+	attrStartTimestamp = "StartTimestamp" // Primary key: when the bucket started (unique identifier)
+	attrPayloadType    = "PayloadType"    // Artificial partition key for Global Secondary Index queries (always "Payload")
```

Comment:
It's kinda confusing how you have `PayloadType` -> `Payload` as a placeholder, and then `Payload` -> `data` for the actual meaningful data

What do you think about using `ArtificialPartitionKey` -> `ArtificialPartitionValue` for the meaningless boilerplate, to not clutter the `Payload` namespace?